app/grandchallenge/algorithms/models.py

```diff
@@ -69,6 +69,38 @@
 JINJA_ENGINE = sandbox.ImmutableSandboxedEnvironment()
 
 
+class AlgorithmInterfaceManager(models.Manager):
+    def get_existing_interface_for_inputs_and_outputs(
+        self, *, inputs, outputs
+    ):
+        try:
+            return self.annotate(
+                input_count=Count("inputs", distinct=True),
+                output_count=Count("outputs", distinct=True),
+            ).get(
+                inputs__in=inputs,
+                outputs__in=outputs,
+                input_count=len(inputs),
+                output_count=len(outputs),
+            )
+        except ObjectDoesNotExist:
+            return None
+
+
+class AlgorithmInterface(models.Model):
+    inputs = models.ManyToManyField(
+        to=ComponentInterface, related_name="inputs"
+    )
+    outputs = models.ManyToManyField(
+        to=ComponentInterface, related_name="outputs"
+    )
```

Comment:
All `ManyToManyField`s should implement a through table. I think this is in our code guidelines, if not, please add it.

Reply:
It wasn't in the guidelines yet. I have added it to our guidelines.